### PR TITLE
[ds-sam-calc] Bond-capped winners: advise bond top-up, not "raise bid"

### PR DESCRIPTION
**Pull request by Cerbah**
## Problem

On the PSR dashboard, a **winning** validator whose SAM stake is capped by its **bond** — not its bid — is told to **"Raise bid to grow stake next epoch."** Raising the bid can't help: the target is pinned at the bond ceiling, so the only lever that grows their stake is a **bond top-up**.

Reported by Cerba on `psr.marinade.finance`. The Next Step / tip text is produced by `getValidatorTip` in this package (`@marinade.finance/ds-sam-calc`), which the dashboard re-exports — so the fix belongs here, not in psr-dashboard.

**Reproduces on (epoch 1003):**
- `5cwxevV9u1poNgfc1tKxMfEGX4oKt1DZNSHg48Umubm3` (bidding ~7.60% APY, already winning)
- `3jkJVgfz1zrHSy6YLK6g96eTj49kCnDj2i8AbbKLZhkk`
- `9FZWpUMfXZ3993g2BfqSFg7xcx9iUCxQwKeYzr2WQCM1`

For the first, simulating ~135 SOL added to the bond lifts `marinadeSamTargetSol` up to `maxStakeWanted` — confirming the bond is the binding constraint.

## Root cause

`bondCta` only fires when bond **health** (runway) is `critical`/`watch`. But runway is measured against the target the bond has **already clamped down** — so a bond that caps growth still looks "healthy," `bondCta` returns `null`, and the tip falls through to `deltaCta`, whose trajectory fallback emits `RAISE_TO_GROW` ("Raise bid to grow stake next epoch.") for any in-set validator below the priority frontier. That misattributes the growth lever to the bid.

## Fix

Add `bondGrowthCta`, returned from `bondCta`'s healthy/non-firing path for an **in-set** validator whose target sits at the **bond-supported** ceiling below their wanted stake:

> **Top up bond to reach your `maxStakeWanted`.**  _(constraint: `bond`, urgency: `info`)_

It outranks `deltaCta`'s `RAISE_TO_GROW` (`rank`/`info`) on the `LEVER_ORDER` tiebreak, so the actionable lever wins. It fires **only** when the BOND is the binding half of `maxBondDelegation` (`min(bondSamStakeCapSol, per-validator TVL cap)`), i.e. `bondSamStakeCapSol <= maxBondDelegation`:
- **TVL cap** binds instead → left to the existing per-validator cap CTA (a top-up wouldn't help).
- **maxStakeWanted** binds → the existing `At your `maxStakeWanted` setting` CTA already covers it.
- bond below-min / unhealthy runway → unchanged (higher-severity CTAs still win).
- `maxStakeWanted` null → treated as "no self-imposed ceiling" and worded "Top up bond to grow stake."

## Tests

`packages/ds-sam-calc/test/tip-engine.test.ts` — 3 new cases:
1. bond ceiling below `maxStakeWanted` → `bond`/`info` "reach your `maxStakeWanted`" (was `RAISE_TO_GROW`)
2. TVL cap is the binder (`bondSamStakeCapSol > maxBondDelegation`) → stays "raise bid", no bond CTA
3. bond ceiling == `maxStakeWanted` → at own cap, no bond CTA

Full `ds-sam-calc` suite green (78 passed); `tsc --build` and prettier clean; no new eslint complexity warning.

## Notes for reviewers
- **Precedence:** this bond `info` CTA also wins over `deltaCta`'s positive "N SOL arriving next epoch" while a bond-capped validator is still filling toward its ceiling. That seems right (top-up is the actionable next step), but flagging in case you'd rather defer it while stake is actively arriving, as the watch-tier CTAs do.
- The CTA text is figure-free (unlike the other `Top up …` strings, which carry a SOL amount). Computing the exact top-up to lift the ceiling to `maxStakeWanted` means inverting `bondStakeCapSam`; happy to add it if you want the amount inline.
- Needs a `ds-sam-calc` version bump + republish for psr-dashboard to pick it up (currently pinned to `0.1.1`).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved in-set validator guidance when bond limits clamp the effective SAM target.
  * Added bond-specific advisory CTAs when the validator can’t reach their `maxStakeWanted` due to the bond cap; avoids suggesting bond top-ups when TVL limits or the validator’s own maximum are the real constraint.
  * Updated messaging so the next step is correctly shown as bond top-up vs “raise bid” depending on which cap applies.
* **Tests**
  * Extended regression coverage for in-set “bond top-up vs raise-bid” scenarios and edge cases.

### CODE REVIEW

File: packages/ds-sam-calc/src/tip-engine.ts [hunk 1]
```diff
@@ -274,7 +274,11 @@ function bondCta(
   // 'watch' implies runway > minBondEpochs + BOND_URGENT_EPOCHS, so any 'watch'
   // validator is already above the fee threshold.
   const fires = health === 'critical' || (inSet && health === 'watch' && (coverage.topUpToKeepStake > 0 || delta <= 0))
-  if (!fires) return null
+  // Bond runway looks fine — but "fine" is measured against a target the bond
```
Comment (janlegner) [inline packages/ds-sam-calc/src/tip-engine.ts]: Refactor the code to not include if-do statements on a single line, always use curly braces and multiline formatting

File: packages/ds-sam-calc/src/tip-engine.ts [hunk 2]
```diff
@@ -373,6 +377,49 @@ function isDefending(validator: AugmentedAuctionValidator, delta: number): boole
   return (validator.marinadeActivatedStakeSol ?? 0) > NON_TRIVIAL_STAKE_SOL && delta < -NON_TRIVIAL_LOSS_SOL
 }
 
+// The BOND — not the bid — is the growth lever when the auction has clipped the
```
Comment (janlegner) [inline packages/ds-sam-calc/src/tip-engine.ts]: reduce the comments to only explain why as a single line, do not duplicate the code in the comment

File: packages/ds-sam-calc/src/tip-engine.ts (hunk 1 above)
Comment (Cerbah) [reply packages/ds-sam-calc/src/tip-engine.ts]: Done — braced all single-line ifs and moved them onto multiple lines. Applied across the whole file, not just the new code: also the pre-existing ones in `tip`, `outOfSetCta` and `capCta`. Pushed in 112fe51.

File: packages/ds-sam-calc/src/tip-engine.ts (hunk 2 above)
Comment (Cerbah) [reply packages/ds-sam-calc/src/tip-engine.ts]: Done — cut the verbose blocks down to a single line explaining the why, no longer restating the code. Reduced the `bondGrowthCta` doc comment, the `bondCta` fall-through note, and the in-function guard comment. Pushed in 112fe51.

**Review by ochaloup — APPROVED** [review]
To me it seems ok now - approving. After merging we need to bump package version to be deployed to npm registry.

I may be able to publish in the evening or @kronael maybe.

**Comment by ochaloup** [comment]
https://github.com/marinade-finance/ds-sam/pull/127
